src/hooks/useFilterPresets/useFilterPresets.ts

```diff
@@ -0,0 +1,114 @@
+// @ts-strict-ignore
+import { SaveFilterTabDialogFormData } from "@dashboard/components/SaveFilterTabDialog";
+import useNavigator from "@dashboard/hooks/useNavigator";
+import {
+  getActiveTabIndexAfterTabDelete,
+  getNextUniqueTabName,
+} from "@dashboard/products/views/ProductList/utils";
+import { StorageUtils } from "@dashboard/utils/filters";
+import { prepareQs } from "@dashboard/utils/filters/qs";
+import { stringify } from "qs";
+import { useState } from "react";
+
+export const useFilterPresets = ({
+  params,
+  reset,
+  storageUtils,
+  getUrl,
+}: {
+  params: Record<string, unknown>;
+  reset: () => void;
+  getUrl: () => string;
+  storageUtils: StorageUtils<string>;
+}) => {
+  const navigate = useNavigator();
+  const baseUrl = getUrl();
```

Comment:
**nitpick**: wouldn't it be better to just pass `baseUrl` string as parameter to this hook?